src/coreclr/System.Private.CoreLib/src/System/RuntimeType.BoxCache.cs

```diff
@@ -69,7 +69,7 @@ private BoxCache(RuntimeType rt)
                 {
                     // If the allocator is null, then we shouldn't allocate and make a copy,
                     // we should return the data as the object it currently is.
-                    return Unsafe.As<byte, object>(ref data);
+                    return Unsafe.ReadUnaligned<object>(ref data);
```

Comment:
presumably, gc references are not expected to be not-aligned

Reply:
I think we can introduce an internal helper in `Unsafe`: `ReadAligned` that checks alignment in Debug.Assert